python/lsst/ts/MTAirCompressor/aircompressor_csc.py

```diff
@@ -150,24 +99,66 @@ def add_arguments(cls, parser: argparse.ArgumentParser) -> None:
     def add_kwargs_from_args(
         cls, args: argparse.Namespace, kwargs: typing.Dict[str, typing.Any]
     ) -> None:
-        """Process custom --hostname and --unit arguments."""
+        """Process custom --hostname, --port and --unit arguments."""
         cls.hostname = (
             f"m1m3cam-aircomp{kwargs['index']:02d}.cp.lsst.org"
             if args.hostname is None
             else args.hostname
         )
+        cls.port = args.port
         cls.unit = kwargs["index"] if args.unit is None else args.unit
 
     async def close_tasks(self) -> None:
+        await super().close_tasks()
         if self.simulation_mode == 1:
             await self.simulator.shutdown()
             await self.simulator_future.cancel()
-        await super().close_tasks()
-
-    async def begin_start(self, data):
+        self.poll_task.cancel()
+        await self.disconnect()
+
+    async def log_modbus_error(self, modbus_error, msg="", ignore_timeouts=False):
+        await self.disconnect()
+
+        if ignore_timeouts is False:
+            if self.summary_state != salobj.State.FAULT and (
+                self._failed_time is None
+                or monotonic() < self._failed_time + self.grace_period
+            ):
+                self.log.error(str(modbus_error))
+                if self._failed_time is None:
+                    self.log.warning(
+                        "Lost compressor connection, will try to reconnect for"
+                        f" {self.grace_period} seconds"
+                    )
+                    self._failed_time = monotonic()
+                return
+
+        try:
+            await self.fault(modbus_error.exception.original_code, msg)
+        except AttributeError:
+            if isinstance(
+                modbus_error.exception, pymodbus.exceptions.ConnectionException
+            ):
+                await self.fault(ErrorCode.COULD_NOT_CONNECT, msg + str(modbus_error))
+            else:
+                await self.fault(ErrorCode.MODBUS_ERROR, msg + str(modbus_error))
+
+        self._failed_time = None
+
+    async def connect(self):
```

Comment:
Please consider resetting failed_time to None here (or, perhaps, in `handle_summary_state`, but I don't think you are using that), in order to make sure that it is always reset when you connect.